android/src/main/java/com/novoda/notils/devicedetection/X86Detector.java

```diff
@@ -0,0 +1,45 @@
+package com.novoda.notils.devicedetection;
+
+import android.annotation.TargetApi;
+import android.os.Build;
+
+import java.util.Arrays;
+
+class X86Detector implements Detector {
+
+    private static final String X86 = "x86";
+
+    private final String[] supportedAbis;
+
+    public static Detector newInstance() {
+        AndroidVersion androidVersion = AndroidVersion.newInstance();
+        if (androidVersion.is21LollipopOrOver()) {
+            return new X86Detector(getSupportedAbis());
+        } else {
+            return new X86Detector(getLegacySupportedAbis());
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    private static String[] getSupportedAbis() {
+        return Build.SUPPORTED_ABIS;
+    }
+
+    private static String[] getLegacySupportedAbis() {
+        return new String[]{Build.CPU_ABI, Build.CPU_ABI2};
+    }
+
+    X86Detector(String... supportedAbis) {
+        this.supportedAbis = Arrays.copyOf(supportedAbis, supportedAbis.length);
```

Comment:
PMS/findbugs will flag it here again if you move the copy to the ctor (not sure if it will flag it if the ctor is private). This ctor can still be used for testing - the values will still be same, you're not gonna mock a string array.